blueocean-dashboard/src/main/js/components/RunMessageCell.jsx

```diff
@@ -11,17 +11,31 @@ export default class RunMessageCell extends Component {
         const run = this.props.run;
         const t = this.props.t;
         let message;
-        if (run && run.description) {
+
+        // Note that the order that this is evaluated is important for providing a relevant message to the user
+
+        // 1. If the user has set a message then we always show it
+        const showUserDefinedMessage = run && run.description;
+
+        // 2. Show commit messages if available
+        //    however if a run has > 1 cause then the cause is likely more important than the change set (e.g. replay)
+        const showCommitMessage = run && run.changeSet && (run.changeSet && run.changeSet.length > 0) && (run.causes && run.causes.length <= 1);
+
+        // 3. Lastly if there are any causes, display the last cause available
+        const showCauses = run && (run.causes && run.causes.length > 0);
```

Comment:
This is well-commented. Would it be simpler though just to check `run.causes.length > 1` and move the `showCauses` check below up above the `showCommitMessage` case? This is a nit, so if that seems silly, please ignore.